torchci/pages/api/flaky-tests/disable.ts

```diff
@@ -0,0 +1,157 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import * as urllib from "urllib";
+
+import { getOctokit } from "lib/github";
+import fetchFlakyTests from "lib/fetchFlakyTests";
+import { FlakyTestData, IssueData } from "lib/types";
+import { supportedPlatforms } from "lib/bot/verifyDisableTestIssueBot";
+import fetchIssuesByLabel from "lib/fetchIssuesByLabel";
+
+
+const NUM_HOURS = 3;
+const owner: string = "pytorch";
+const repo: string = "pytorch";
+
+
+export default async function handler(
+    req: NextApiRequest,
+    res: NextApiResponse<void>
+) {
+    await disableFlakyTests();
+    res.status(200);
+}
+
+
+async function disableFlakyTests() {
+    const octokit = await getOctokit(owner, repo);
+    const flaky_tests = await fetchFlakyTests(`${NUM_HOURS}`);
+    const issues: IssueData[] = await fetchIssuesByLabel("skipped");
+    flaky_tests.forEach(async function(test: FlakyTestData) {
+        const issueTitle = getIssueTitle(test.name, test.suite);
+        const matchingIssues = issues.filter((issue) => issue.title === issueTitle);
+        if (matchingIssues.length !== 0) {
+            // There is a matching issue
+            const matchingIssue = matchingIssues[0];
+            if (matchingIssue.state === "open") {
+                const body = `Another case of flakiness has been found [here](${getLatestWorkflowURL(test.workflow_ids)}).
+                Please verify the issue was opened after this instance, that the platforms list includes all of
+                [${getPlatformsAffected(test.workflow_names).join(", ")}], or disable bot might not be working as expected.`;
+                octokit.rest.issues.createComment({
+                    owner,
+                    repo,
+                    issue_number: matchingIssue.number,
+                    body,
+                });
+            } else {
+                // Re-open the issue
+                octokit.rest.issues.update({
+                    owner,
+                    repo,
+                    issue_number: matchingIssue.number,
+                    state: "open",
+                });
+
+                const body = `Another case of flakiness has been found [here](${getLatestWorkflowURL(test.workflow_ids)}).
+                Reopening the issue to disable. Please verify that the platforms list includes all of
```

Comment:
what does this verify platform list mean? Like does a single workflow get run on multiple platforms (probably a dumb question).

Reply:
yes, one test can run across multiple platforms (where a platform is a given operating system and/or hardware target)